rfc-136-remove-our-backup-cdn-in-gcp.md

```diff
@@ -0,0 +1,130 @@
+# Remove the backup CDN in Google Cloud Platform (GCP)
+
+## Summary
+
+We have two backup content delivery networks (CDNs), in case Fastly have an outage.
+The first is AWS CloudFront, the second is Google Cloud CDN.
+
+A simultaneous outage of both Fastly and AWS CloudFront is very unlikely. So
+the Google Cloud CDN backup provides very little extra reliability.
+
+At the same time, the cost of maintaining the Google Cloud CDN backup is
+non-trivial.
+
+The maintenance cost outweighs the expected reliability benefit, so we should
+remove the Google Cloud CDN backup.
+
+## Problem
+
+GOV.UK's current business continuity plan aims to ensure that members of the
+public can continue to use www.gov.uk, even in the event of a major outage in
+one or more of our infrastructure providers.
+
+Firstly, while our primary CDN (Fastly) is up, any failed requests to our origin servers
+will be retried against three static mirrors - two in AWS, one in GCP.
+
+* origin
+* AWS S3
+* AWS S3 (replica in a different region)
+* Google Cloud Storage
+
+If none of the above are successful, an error is served to the user.
+
+Secondly, if our primary CDN (Fastly) is down, we have two backup CDNs which we
+can manually fail over to (by updating a DNS record). AWS CloudFront and Google
+Cloud CDN.
+
+The mirroring approach works well. This RFC does not propose any change to the
+mirrors, including the Google Cloud Storage mirror.
+
+The AWS CloudFront CDN is reasonably easy to maintain and test, however the
+Google Cloud CDN backup is significantly more difficult.
+
+### Our Google Cloud CDN setup currently broken
+
+The certificate in use on the CDN expired in May 2020.
+
+The Google Cloud Storage bucket the CDN is pointing to is currently private.
```

Comment:
This is intended behaviour to save money in case someone/thing decides to crawl against the Google bucket. As part of the playbook for activating the Google CDN, we would mark the bucket as public and re-point the www.gov.uk DNS to the Google Cloud CDN fronting the bucket.

The main issue we have is the certificate issuing before the live traffic switch.

Reply:
It's interesting that we don't bother with this for the CloudFront / S3 backup. What's to stop something from crawling that?

In any case, I think it's still worth mentioning this as something that's broken. If we had documentation on how to fail over which included making the bucket public, that would be fine. But we don't.

Reply:
Umm.. my understanding is that the S3 buckets should be protected, see here: https://github.com/alphagov/govuk-aws/blob/master/terraform/projects/infra-mirror-bucket/mirror-read-policy.tf 

I don't know what policy they had on the Cloudfont side as I didn't do that work. My thoughts were they were going to be only office IPs allowed to use the Cloudfront endpoint using test and we widen the whitelist once it is made operational.